docs/development/pigs/pig-026.rst

```diff
@@ -0,0 +1,287 @@
+  norm.prior = prior
+  
+  spectral_model = PowerLawSpectraModel() * norm
+
+Additional use case:  Add sigma v estimator functionality for dark matter use case `#2075`_
+
+Implementation
+==============
+
+The following implementation draft is compatible with the current API,
+where the models are set via Dataset.model and the Fit is run via Fit.run(datasets).
+
+.. code:: python
+
+  class PriorModel(ModelBase):
+
+        _weight = 1
+        _type = "prior"
+
+        @property
+        def parameters(self):
+            """PriorParameters (`~gammapy.modeling.PriorParameters`)"""
+            return PriorParameters(
+                [getattr(self, name) for name in self.default_parameters.names]
+            )
+                    
+                  
+        @property
+        def weight(self):
+            return self._weight
+        
+        @weight.setter
+        def weight(self, value):
+            self._weight = value
+            
+        def __call__(self, value):
+            """Call evaluate method"""
+            kwargs = {par.name: par.quantity for par in self.parameters}
+            if isinstance(value, Parameter):
+                return self.evaluate(value.quantity, **kwargs)     
+             else:
+                raise TypeError(f"Invalid type: {value}, {type(value)}")
+            
+
+The base ``PriorModel`` class inherits from the ``ModelBase`` class and allows to set (for now one) unit to which the to-be-evaluated parameter the prior is set on is converged. In addition, there can be a weight set. The parameters of the  ``PriorModel`` are ``PriorParameters`` and ``PriorParameter``. They inherit from the ``Parameters`` and ``Parameter`` classes, respectively, however, have a limited amount of attributes. 
+It is assumed that the ``PriorParameters`` are set in the same unit as the ``Parameter`` they get evaluated on.  
+
+Future additional properties of the ``PriorModel`` classes: 
+
+- Write/read from/to a yaml file, ideally also when the corresponding model is written/read (see serialisation example below)
+- Prior registry system
+- Different prior subclasses depending on the use cases 
+
+Exemplary additional prior subclasses:
+---------------------------------------
+.. code:: python
+
+  class GaussianPrior(PriorModel):
+
+        """Gaussian Prior with mu and sigma.
+        """
+        tag = ["GaussianPrior"]
+        _type = "prior"
+        mu = PriorParameter(name="mu", value = 0, unit = '')
+        sigma = PriorParameter(name="sigma", value = 1, unit = '')
+        
+        @staticmethod
+        def evaluate(value, mu, sigma):
+            return ((value - mu) / sigma) ** 2
+
+
+.. code:: python
+
+  class UniformPrior(PriorModel):
+
+        """Uniform Prior
+        """
+        tag = ["UniformPrior"]
+        uni = PriorParameter(name="uni", value = 0, min = 0, max = 10, unit = '' )
+        
+        @staticmethod
+        def evaluate(value, uni):
+            return uni
+
+            
```

Comment:
A prior that can be quite useful is the LogUniform prior.
For example for the normalization of a source, giving a linear prior between 1e-14 and 1e-11 is not ideal. Having the LogUniform priors will become handy if at some point we perform Bayesian analyses.
It could be something like this : 

```
class LogUniformPrior(PriorModel):
         """LogUniform Prior
         """
         tag = ["LogUniformPrior"]
         uni = PriorParameter(name="loguni", value = 1, min = 0.1, max = 10, unit = '' )
```

A note on the fact that in the Uniform and LogUniform priors the `value` parameter itself is a not useful as we just want to define (min, max) boundaries.
Is there a reason we should have a `value` parameter in this case ?

Reply:
Thanks for the comment! 
Yes, I think this kind of prior could be helpful and is very straightforward to implement. 
The reason for the ``value`` parameter was to gain more flexibility in the use cases. In some cases, one might the prior to be stronger than in others.

Reply:
Does this prior need a range for the parameter , e.g.:
```
 min = PriorParameter(name="min", value = -np.inf, unit = '' )
 max = PriorParameter(name="min", value = np.inf, unit = '' )
```